firmware/baseband/proc_adsbrx.cpp

```diff
@@ -31,29 +33,27 @@
 using namespace adsb;
 
 void ADSBRXProcessor::execute(const buffer_c8_t& buffer) {
-    int8_t re, im;
-    uint32_t mag;
-    uint32_t c;
-    uint8_t bit, byte{};
-
     // This is called at 2M/2048 = 977Hz
-    // One pulse = 500ns = 2 samples
-    // One bit = 2 pulses = 1us = 4 samples
+    // Each sample is 500ns.
+    // One bit is 2 samples == 1us.
+    // Bit value is the transition between samples.
+    // i.e. lo->hi == 0, hi->lo == 1
 
     if (!configured) return;
 
+    uint8_t bit = 0;
+    uint8_t byte = 0;
```

Comment:
Byte should really be a member and it should be reset whenever bit_count is reset?
I suppose it will have shifted off all stale bits in the later case, but it seems possible that the bytes don't end up lining up with the sample buffers. That said there are 128 bytes per buffer (ish.. preamble kind of throws things off).